framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -4240,6 +4291,202 @@ private void makeGroundTargetType(
         addDefaultAnnotations(functionalType);
     }
 
+    /**
+     * Return true if {@code type} should be captured.
+     *
+     * <ul>
+     *   <li>{@code type} and {@code typeMirror} must both be declared types.
+     *   <li>{@code type} must not be a raw type, must not have an uninferred type argument, and
+     *       must have a wildcard as a type argument.
+     *   <li>{@code typeMirror} must have a captured type as a type argument.
+     * </ul>
+     *
+     * @param type annotated type to capture
+     * @param typeMirror underlying Java type
+     * @return true if {@code type} should be captured
+     */
+    private boolean shouldCapture(AnnotatedTypeMirror type, TypeMirror typeMirror) {
+        if (type.getKind() != TypeKind.DECLARED && typeMirror.getKind() != TypeKind.DECLARED) {
+            return false;
+        }
+        DeclaredType capturedTypeMirror = (DeclaredType) typeMirror;
+        AnnotatedDeclaredType typeToCapture = (AnnotatedDeclaredType) type;
+        if (typeToCapture.getTypeArguments().isEmpty()
```

Comment:
The first and fourth disjuncts logically belong to the third bullet point in the Javadoc.  Please make this `if` statement be just about the second bullet point, and put the first and fourth disjuncts afterward, together with the `for` statement that addresses the third bullet point.